scout/server/blueprints/variants/controllers.py

```diff
@@ -445,47 +444,77 @@ def variant_export_lines(store, case_obj, variants_query):
+        if not tx.get("refseq_identifiers") and tx.get("is_primary") is False:
+            continue
+
+        for var_tx in variant_gene.get("transcripts", []):
+            if var_tx["transcript_id"] != tx_id:
+                continue
+
+            tx_refseq = tx.get("refseq_id")
+            hgvs = var_tx.get("coding_sequence_name") or "-"
+            pt_change = var_tx.get("protein_sequence_name") or "-"
+
+            # collect info from primary transcripts
+            if tx_refseq in hgnc_gene.get("primary_transcripts"):
+                primary_txs.append("|".join([tx_refseq or tx_id, hgvs, pt_change]))
+
+            # collect info from canonical transcript
+            if tx_id == variant_gene.get("canonical_transcript"):
+                canonical_txs.append("|".join([tx_refseq or tx_id, hgvs, pt_change]))
+
+    return canonical_txs, primary_txs
+
+
+def variant_export_genes_info(store, gene_list, genome_build="37"):
     """Adds gene info to a list of fields corresponding to a variant to be exported.
 
     Args:
-        gene_list(list) A list of gene objects contained in the variant
+        gene_list(list) A list of gene objects (with limited info) contained in the variant
+        genome_build(str): genome build to export gene list to
 
     Returns:
         gene_info(list) A list of gene-relates string info
     """
     gene_ids = []
     gene_names = []
     canonical_txs = []
-    hgvs_c = []
-    pt_c = []
-
+    primary_txs = []
     gene_info = []
 
     for gene_obj in gene_list:
         hgnc_id = gene_obj["hgnc_id"]
-        gene_name = gene(store, hgnc_id)["symbol"]
-
         gene_ids.append(hgnc_id)
-        gene_names.append(gene_name)
-
-        if gene_obj.get("canonical_transcript"):
-            canonical_txs.append(gene_obj.get("canonical_transcript"))
+        hgnc_gene = store.hgnc_gene(hgnc_id, genome_build)
+        if hgnc_gene is None:
+            continue
+        gene_names.append(hgnc_gene.get("hgnc_symbol"))
 
-        hgvs_nucleotide = "-"
-        protein_change = "-"
-        # gather HGVS info from gene transcripts
+        gene_canonical_txs, gene_primary_txs = match_gene_txs_variant_txs(gene_obj, hgnc_gene)
 
-        transcripts_list = gene_obj.get("transcripts")
-        for transcript_obj in transcripts_list:
-            if transcript_obj["transcript_id"] == gene_obj.get("canonical_transcript"):
-                hgvs_nucleotide = transcript_obj.get("coding_sequence_name") or "-"
-                protein_change = transcript_obj.get("protein_sequence_name") or "-"
-        hgvs_c.append(hgvs_nucleotide)
-        pt_c.append(protein_change)
+        canonical_txs += gene_canonical_txs
+        primary_txs += gene_primary_txs
 
-    for item in [gene_ids, gene_names, canonical_txs, hgvs_c, pt_c]:
-        gene_info.append(";".join(str(x) for x in item))
+    for item in [gene_ids, gene_names, canonical_txs, primary_txs]:
+        gene_info.append("-".join(str(x) for x in item))
```

Comment:
I think you need to join with `" - "` (with spaces). Since gene symbols can contain dashes things like `CYP1B1-CYP1B1-AS1` become ambiguous.